maze_transformer/training/tokenizer.py

```diff
@@ -60,3 +71,92 @@ def as_tokens(
             )
 
         return tokens
+
+
+class HuggingMazeTokenizer(PreTrainedTokenizer):
+    vocab: dict[str, int]  # map of token_ids to strings
+
+    bos_token: str = SPECIAL_TOKENS["start_path"]
+    eos_token: str = SPECIAL_TOKENS["end_path"]
+    pad_token: str = SPECIAL_TOKENS["padding"]
+    unk_token: str = "<UNK>"
+
+    vocab_size: int = 0
```

Comment:
is this supposed to always be 0?

Reply:
Yes - when we use the `PreTrainedTokenizer`'s `add_token...` methods, those "added tokens" are considered an additional part of the "base vocab" which would typically be loaded from a file. We circumvent the need to have a file containing our tokens by having an empty base vocab, such that everything is loaded during initialization from the maze config